tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java

```diff
@@ -152,31 +150,25 @@ public Connection getConnection(ChannelId channelId) {
     }
 
     public void close() {
-        for (ChannelId id : connections.keySet()) {
-            Connection conn = connections.get(id);
-            if (conn != null) {
-                conn.close();
-            }
+        for ( Connection conn : connections.values()) {
+            conn.close();
         }
-
         this.connections.clear();
+
     }
 
     public Map<String, Integer> getStats() {
         int in = 0;
         int out = 0;
-        for (ChannelId id : connections.keySet()) {
-            Connection conn = connections.get(id);
-            if (conn == null) {
-                continue;
-            } else if (conn.direction == Connection.Direction.OUT) {
+        for (Connection conn : connections.values()) {
+            if (conn.direction == Connection.Direction.OUT) {
                 out++;
             } else {
                 in++;
             }
         }
 
-        Map<String, Integer> result = new HashMap<>();
+        Map<String, Integer> result = new HashMap<>(2);
```

Comment:
Unfortunately, passing `2` to the constructor won't allow the map to hold 2 entries without re-sizing (have to account for the load factor, which is `0.75` by default). Has to be something like:
```
public static int hashMapCapacity(int size) { return (size * 4 + 2) / 3; } // for load factor of 0.75
...
   new HashMap(hashMapCapacity(2), 0.75f);
```